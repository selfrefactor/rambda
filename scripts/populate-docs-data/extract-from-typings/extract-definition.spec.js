import { extractDefinition } from './extract-definition'

test('happy', () => {
  expect(extractDefinition()).toMatchSnapshot()
})
