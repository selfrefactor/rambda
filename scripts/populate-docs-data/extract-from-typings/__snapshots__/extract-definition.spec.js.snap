// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`happy 1`] = `
Object {
  "F": "F(): boolean",
  "T": "T(): boolean",
  "add": "add(a: number, b: number): number",
  "adjust": "adjust<T>(index: number, replaceFn: (a: T) => T, list: ReadonlyArray<T>): T[]",
  "all": "all<T>(predicate: (x: T) => boolean, list: ReadonlyArray<T>): boolean",
  "allPass": "allPass<T>(predicates: ((x: T) => boolean)[]): (input: T) => boolean",
  "always": "always<T>(x: T): () => T",
  "and": "and<T extends { and?: ((...a: readonly any[]) => any)",
  "any": "any<T>(predicate: (x: T, i: number) => boolean, list: ReadonlyArray<T>): boolean",
  "anyPass": "anyPass<T>(predicates: ReadonlyArray<SafePred<T>>): SafePred<T>",
  "append": "append<T>(x: T, listOrString: ReadonlyArray<T>): T[]",
  "applySpec": "applySpec<Spec extends Record<string, (...args: readonly any[]) => any>>(
  conditions: Spec
): (
  ...args: Parameters<ValueOfRecord<Spec>>
) => { [Key in keyof Spec]: ReturnType<Spec[Key]> }",
  "assoc": "assoc<T, U, K extends string>(prop: K, newValue: T, obj: U): Record<K, T> & U",
  "assocPath": "assocPath<T, U>(path: Path, newValue: T, obj: U): U",
  "both": "both(pred1: Pred, pred2: Pred): Pred",
  "chain": "chain<T, U>(fn: (n: T) => readonly U[], list: readonly T[]): U[]",
  "clamp": "clamp(min: number, max: number, input: number): number",
  "clone": "clone<T>(input: T): T",
  "complement": "complement(pred: (...args: any[]) => boolean): (...args: any[]) => boolean",
  "compose": "compose<T1>(fn0: () => T1): () => T1",
  "concat": "concat<T>(x: ReadonlyArray<T>, y: ReadonlyArray<T>): T[]",
  "cond": "cond(conditions: [Pred, (...a: readonly any[]) => any][]): (...a: readonly any[]) => any",
  "converge": "converge(after: ((...a: readonly any[]) => any), fns: Array<((...a: readonly any[]) => any)>): (...a: readonly any[]) => any",
  "curry": "curry<F extends (...args: any) => any>(f: F): FunctionToolbelt.Curry<F>",
  "curryN": "curryN(length: number, fn: (...args: readonly any[]) => any): (...a: readonly any[]) => any",
  "dec": "dec(x: number): number",
  "defaultTo": "defaultTo<T>(defaultValue: T): (...inputArguments: (T | null | undefined)[]) => T",
  "difference": "difference<T>(a: ReadonlyArray<T>, b: ReadonlyArray<T>): T[]",
  "dissoc": "dissoc<T>(prop: string, obj: any): T",
  "divide": "divide(a: number, b: number): number",
  "drop": "drop<T>(howManyToDrop: number, listOrString: ReadonlyArray<T>): T[]",
  "dropLast": "dropLast<T>(howManyToDrop: number, listOrString: ReadonlyArray<T>): T[]",
  "either": "either(firstPredicate: Pred, secondPredicate: Pred): Pred",
  "endsWith": "endsWith(target: string, str: string): boolean",
  "equals": "equals<T>(a: T, b: T): boolean",
  "filter": "filter<T>(predicate: FilterFunctionArray<T>): (x: T[]) => T[]",
  "find": "find<T>(predicate: (a: T) => boolean, arr: ReadonlyArray<T>): T | undefined",
  "findIndex": "findIndex<T>(findFn: (a: T) => boolean, arr: ReadonlyArray<T>): number",
  "findLast": "findLast<T>(fn: (a: T) => boolean, list: T[]): T | undefined",
  "findLastIndex": "findLastIndex<T>(fn: (a: T) => boolean, list: T[]): number",
  "flatten": "flatten<T>(x: ReadonlyArray<T> | ReadonlyArray<T[]> | ReadonlyArray<ReadonlyArray<T>>): T[]",
  "flip": "flip<T, U, TResult>(fn: (arg0: T, arg1: U) => TResult): (arg1: U, arg0?: T) => TResult",
  "forEach": "forEach<T>(fn: (x: T) => void, list: T[]): T[]",
  "fromPairs": "fromPairs<V>(listOfPairs: KeyValuePair<string, V>[]): { [index: string]: V }",
  "groupBy": "groupBy<T>(groupFn: (a: T) => string, list: ReadonlyArray<T>): { [index: string]: T[] }",
  "groupWith": "groupWith<T>(compareFn: (x: T, y: T) => boolean): (list: ReadonlyArray<T>) => T[][]",
  "has": "has<T>(prop: string, obj: T): boolean",
  "hasPath": "hasPath<T>(
  path: string | string[],
  input: object
): boolean",
  "head": "head<T>(listOrString: T[]): T | undefined",
  "identical": "identical<T>(a: T, b: T): boolean",
  "identity": "identity<T>(input: T): T",
  "ifElse": "ifElse(condition: Pred, onTrue: Arity1Fn, onFalse: Arity1Fn): Arity1Fn",
  "inc": "inc(x: number): number",
  "includes": "includes(valueToFind: string, input: ReadonlyArray<string> | string): boolean",
  "indexBy": "indexBy<T>(condition: (x: T) => string, list: ReadonlyArray<T>): { [key: string]: T }",
  "indexOf": "indexOf<T>(valueToFind: T, list: ReadonlyArray<T>): number",
  "init": "init<T>(listOrString: ReadonlyArray<T>): T[]",
  "intersection": "intersection<T>(listA: ReadonlyArray<T>, listB: ReadonlyArray<T>): T[]",
  "intersperse": "intersperse<T>(separator: T, list: ReadonlyArray<T>): T[]",
  "is": "is(targetPrototype: any, x: any): boolean",
  "isEmpty": "isEmpty<T>(x: T): boolean",
  "isNil": "isNil(x: any): x is null | undefined",
  "join": "join(x: string, xs: ReadonlyArray<any>): string",
  "keys": "keys<T extends object>(x: T): (keyof T)[]",
  "last": "last<T>(listOrString: T[]): T | undefined",
  "lastIndexOf": "lastIndexOf<T>(target: T, list: ReadonlyArray<T>): number",
  "length": "length<T>(listOrString: ReadonlyArray<T>): number",
  "lens": "lens<T, U, V>(getter: (s: T) => U, setter: (a: U, s: T) => V): Lens",
  "lensIndex": "lensIndex(index: number): Lens",
  "lensPath": "lensPath(path: RamdaPath): Lens",
  "lensProp": "lensProp(prop: string): {
  <T, U>(obj: T): U",
  "map": "map<T, U>(fn: MapFunctionObject<T, U>, list: Dictionary<T>): Dictionary<U>",
  "match": "match(regExpression: RegExp, str: string): any[]",
  "mathMod": "mathMod(x: number, y: number): number",
  "max": "max<T extends Ord>(x: T, y: T): T",
  "maxBy": "maxBy<T>(compareFn: (input: T) => Ord, x: T, y: T): T",
  "mean": "mean(list: ReadonlyArray<number>): number",
  "median": "median(list: ReadonlyArray<number>): number",
  "merge": "merge<T1, T2>(target: T1, newProps: T2): Merge<T2, T1>",
  "min": "min<T extends Ord>(x: T, y: T): T",
  "minBy": "minBy<T>(compareFn: (input: T) => Ord, x: T, y: T): T",
  "modulo": "modulo(x: number, y: number): number",
  "multiply": "multiply(x: number, y: number): number",
  "negate": "negate(x: number): number",
  "none": "none<T>(predicate: (x: T) => boolean, list: ReadonlyArray<T>): boolean",
  "not": "not(input: any): boolean",
  "nth": "nth<T>(index: number, list: ReadonlyArray<T>): T | undefined",
  "omit": "omit<T, K extends string>(propsToOmit: readonly K[], obj: T): Omit<T, K>",
  "over": "over<T>(lens: Lens, fn: Arity1Fn, value: T): T",
  "partial": "partial<V0, V1, T>(fn: (x0: V0, x1: V1) => T, x0: V0): (x1: V1) => T",
  "path": "path<Input, T>(pathToSearch: string | string[], obj: Input): T | undefined",
  "pathOr": "pathOr<T>(defaultValue: T, pathToSearch: Path, obj: any): T",
  "paths": "paths<Input, T>(pathsToSearch: Path[], obj: Input): (T | undefined)[]",
  "pick": "pick<T, K extends string | number | symbol>(propsToPick: readonly K[], input: T): Pick<T, Exclude<keyof T, Exclude<keyof T, K>>>",
  "pickAll": "pickAll<T, U>(propsToPick: ReadonlyArray<string>, obj: T): U",
  "pipe": "pipe<T1>(fn0: () => T1): () => T1",
  "pluck": "pluck<T>(property: number, list: ReadonlyArray<T>): T",
  "prepend": "prepend<T>(x: T, listOrString: ReadonlyArray<T>): T[]",
  "product": "product(list: ReadonlyArray<number>): number",
  "prop": "prop<P extends keyof T, T>(propToFind: P, obj: T): T[P]",
  "propEq": "propEq<T, K extends keyof T>(propToFind: K, valueToMatch: T[K], obj: T): boolean",
  "propIs": "propIs(type: any, name: string, obj: any): boolean",
  "propOr": "propOr<T, U, V>(defaultValue: T, property: string, obj: U): V",
  "range": "range(start: number, end: number): number[]",
  "reduce": "reduce<T, TResult>(reducer: (prev: TResult, current: T, i: number) => TResult, initialValue: TResult, list: ReadonlyArray<T>): TResult",
  "reject": "reject<T>(predicate: FilterFunctionArray<T>): (x: T[]) => T[]",
  "repeat": "repeat<T>(x: T, timesToRepeat: number): T[]",
  "replace": "replace(strOrRegex: RegExp | string, replacer: string, str: string): string",
  "reverse": "reverse<T>(listOrString: ReadonlyArray<T>): T[]",
  "set": "set<T, U>(lens: Lens, replacer: U, obj: T): T",
  "slice": "slice(from: number, to: number, list: string): string",
  "sort": "sort<T>(sortFn: (a: T, b: T) => number, list: ReadonlyArray<T>): T[]",
  "sortBy": "sortBy<T>(sortFn: (a: T) => Ord, list: ReadonlyArray<T>): T[]",
  "split": "split(separator: string | RegExp): (str: string) => string[]",
  "splitEvery": "splitEvery<T>(sliceLength: number, listOrString: ReadonlyArray<T>): T[][]",
  "startsWith": "startsWith(target: string, str: string): boolean",
  "subtract": "subtract(x: number, y: number): number",
  "sum": "sum(list: ReadonlyArray<number>): number",
  "symmetricDifference": "symmetricDifference<T>(x: ReadonlyArray<T>, y: ReadonlyArray<T>): T[]",
  "tail": "tail<T>(listOrString: ReadonlyArray<T>): T[]",
  "take": "take<T>(howMany: number, listOrString: ReadonlyArray<T>): T[]",
  "takeLast": "takeLast<T>(howMany: number, listOrString: ReadonlyArray<T>): T[]",
  "tap": "tap<T>(fn: (a: T) => any, x: T): T",
  "test": "test(regExpression: RegExp): (str: string) => boolean",
  "times": "times<T>(fn: (i: number) => T, howMany: number): T[]",
  "toLower": "toLower(str: string): string",
  "toPairs": "toPairs<S>(obj: { [k: string]: S } | { [k: number]: S }): [string, S][]",
  "toString": "toString<T>(x: T): string",
  "toUpper": "toUpper(str: string): string",
  "transpose": "transpose<T>(list: T[][]): T[][]",
  "trim": "trim(str: string): string",
  "type": "type(x: any): \\"Object\\" | \\"Number\\" | \\"Boolean\\" | \\"String\\" | \\"Null\\" | \\"Array\\" | \\"Function\\" | \\"Undefined\\" | \\"Async\\" | \\"Promise\\" | \\"RegExp\\" | \\"NaN\\"",
  "uniq": "uniq<T>(list: ReadonlyArray<T>): T[]",
  "uniqWith": "uniqWith<T, U>(uniqFn: (x: T, y: T) => boolean, list: ReadonlyArray<T>): T[]",
  "update": "update<T>(index: number, newValue: T, list: ReadonlyArray<T>): T[]",
  "values": "values<T extends object, K extends keyof T>(obj: T): T[K][]",
  "view": "view<T, U>(lens: Lens): (target: T) => U",
  "when": "when<T>(
  rule: Func<boolean>, resultOrFunction: T | IdentityFunction<T>
): IdentityFunction<T>",
  "without": "without<T>(matchAgainst: ReadonlyArray<T>, source: ReadonlyArray<T>): T[]",
  "xor": "xor(x: boolean, y: boolean): boolean",
  "zip": "zip<K, V>(x: ReadonlyArray<K>, y: ReadonlyArray<V>): KeyValuePair<K, V>[]",
  "zipObj": "zipObj<T>(keys: ReadonlyArray<string>, values: ReadonlyArray<T>): { [index: string]: T }",
}
`;
